pymtl3/passes/VcdGenerationPass.py

```diff
@@ -81,6 +79,57 @@ def _gen_vcd_symbol():
         yield code
         n += 1
 
+    # The actual dump_vcd function to be appended to _sched.
+    # The framework will call this function without arguments, and therefore
+    # I chose to monkey patch the data structure objects to the global
+    # namespace of `dump_vcd`. This might be hacky but probably better than
+    # compiling a million lines of code to a Python function (you will almost
+    # certainly get a segfault from that).
+    # TODO: make this less hacky? type check?
+
+    def dump_vcd():
+      s             = top
+      vcd_file      = vcdmeta.vcd_file
+      clock_net_idx = vcdmeta.clock_net_idx
+      clock_symbol  = net_symbol_mapping[vcdmeta.clock_net_idx]
+      next_neg_edge = 100*vcdmeta.sim_ncycles+50
+      next_pos_edge = 100*vcdmeta.sim_ncycles+100
+
+      try:
+        # Dump VCD
+        for i, _net in enumerate( trimmed_value_nets ):
```

Comment:
I think if you do the previous two nested function approach, the eval expression over trimmed_value_nets can be pre-processed in the outer function. Also, thanks for adding this type check thing. I think you can also store last_value in a list or something instead of always doing getattr.

Reply:
What will be the performance benefit of moving the `eval` to the outside function?

Reply:
You are correct. The reason why I needed to generate those statements is to eval the names ..

Reply:
I guess the eval thing can be very slow if it needs to be called every cycle. Did you guys try the 23000 signal vcd again?

Reply:
I remember generating vcd for a large network is observably slower than running just the simulation. There is a lag before the simulation starts (probably recursing the hierarchy and finishing the initialization) and then every cycle takes longer time.